hibernate-core/src/main/java/org/hibernate/engine/internal/StatefulPersistenceContext.java

```diff
@@ -408,7 +408,7 @@ public Object getEntity(EntityKey key) {
 
 	@Override
 	public boolean containsEntity(EntityKey key) {
-		return entitiesByKey == null ? false : entitiesByKey.containsKey( key );
+		return entitiesByKey != null && entitiesByKey.containsKey( key );
```

Comment:
just my personal opinion but not sure this helps code readability

Reply:
I copied the suggestion by Intellij IDE and I think its suggestion does make sense in this case. If you think the code change is unnecessary, I can revert it back.